x-pack/plugins/security_solution/server/usage/collector.ts

```diff
@@ -308,6 +309,1604 @@ export const registerCollector: RegisterCollector = ({
               },
             },
           },
+          detection_rule_status: {
+            all_rules: {
+              eql: {
+                failed: {
+                  type: 'long',
+                  _meta: { description: 'The number of failed rules' },
+                },
```

Comment:
Real bummer this can't be broken up into smaller files and destructured all nicely so it's more easily consumed. Sounds like there's some telemetry tooling that is preventing this? We should open an issue to fix that tooling so these schemas can be better digested.

Do you have more details on this @afharo?

Reply:
I'm also curious if the existing tooling does anything to 'lock' the file. @FrankHassanabad mentioned in another comment below about how it's very important to not _change telemetry keys_, and so am wondering if there is any tooling in place for this? Once broken out into separate files, we could always do a [server-side hook](https://stackoverflow.com/a/27554025/16736698) to lock them in read-only mode perhaps? Same goes with any downstream generated files from these schemas.